src/Services/ShopService/Shopify.php

```diff
@@ -27,31 +31,31 @@ public function __construct(Client $client)
      */
     public function products(array $params = []): array
     {
-        $response = $this->request('get', 'products', '2020-10', $params);
+        $response = $this->request('get', 'products', $params);
 
         return array_map(function ($product) {
-            return [
+            return new Product([
                 'id' => Arr::get($product, 'id'),
                 'title' => Arr::get($product, 'title'),
                 'price' => Arr::get($product, 'variants.0.price')
-            ];
+            ]);
         }, Arr::get($response, 'products'));
     }
 
     /**
      * Retrieves product by id
      * @param int $id
-     * @return array
+     * @return Product
      */
-    public function productById($id): array
+    public function product($id): Product
     {
-        $response = $this->request('get', 'products/' . $id, '2020-10');
+        $response = $this->request('get', 'products/' . $id);
 
-        return [
+        return new Product([
```

Comment:
this logic is now written twices for every collection (orders, products, customers). If we want to add one field to product we have to do it in two places not really handy.

Create private methods:
mapProduct(array $product) {
    return new Product([...])
}

Do this for all collections